packages/core-mobile/app/services/wallet/MnemonicWallet.ts

```diff
@@ -269,8 +231,14 @@ export class MnemonicWallet implements Wallet {
   }: {
     accountIndex: number
     network: Network
-    provider: JsonRpcBatchInternal
+    provider: JsonRpcBatchInternal | Avalanche.JsonRpcProvider
```

Comment:
```suggestion
    provider: JsonRpcBatchInternal
```
You can say:

⸻
Do we still need `Avalanche.JsonRpcProvider` here? If not, I don’t think the `JsonRpcBatchInternal` type guard below is necessary either.